Fix missing header and declaration after goto

th1000s opened the pull request:
gcc < 11 complains "[goto] label can only be part of a statement and a declaration is not a statement" and and clang gets
completely confused: "expected expression"

---

gcc 11 and newer seems to be completely ok with declarations after labels.

rockowitz commented:
Thank you for the patches and the comments.  I was in the process of fixing the "label can only be part of a statement" error addressing issue #280. However I was puzzled as to why user @bmcgruddy was encountering the error and I was not.  I was looking (unsuccessfully, of course) for some gcc configuration setting or make option that explained the difference. Googling the problem hit posts to the effect that gcc should be changed to allow declarations immediately following a statement, but not that in fact gcc had been changed in version 11 to lift the restriction.  See [GCC 11 Series Changes, New Features, and Fixes](https://gcc.gnu.org/gcc-11/changes.html).  Without your comment I'd likely still be searching for the answer as to what was going on.

Re issue templates, your patch instructs the user to submit the output of **ddcutil environment --verbose** in the body of the issue.  The problem with doing this is that the output is so voluminous that it makes the issue thread difficult to follow, which is why I tell users to submit the output as an attachment of some sort. I will consider more suitable issue templates.

I hope you'll forgive my not merging your pull request as a way to acknowledge your input.  I was much simpler to stick with my changes than to cherry pick yours to merge with mine while ignoring the issue template

th1000s commented:
Happy to help, as to this PR, feel free to do as you see fit (though the current dev branch still does not compile for me without the commit here).

Regarding the issue template, that wasn't supposed to be included. I didn't even notice that me testing this (via the web UI) on my clone of the repo also updated this PR, I have removed it now and placed it on a separate `template_test` branch. Have a look at that if you are okay with `.github/` subdir being added to the repo. If found the expandable `<details>..` section trick in some other repo and think that is a good way to include verbose output inline.

rockowitz commented:
Fixed in release 1.4.1
